plugin/plugin.go

```diff
@@ -382,23 +406,151 @@ func (p *AzurePlugin) ListHosts(ctx context.Context, req *pb.ListHostsRequest) (
 						}
 					}
 				}
+				vmToNetworkMap[vmId] = netInfo
+			}
+		}
+	}
+
+	// Now we need to do the same thing for virtual machine scalesets and the VMs they contain
+	// This is slightly different because the VMSS encapsulates the VM instances and their network configurations
+	{
+		var virtualMachineScaleSetTypeSb strings.Builder
+		virtualMachineScaleSetTypeSb.WriteString(constMsComputeService)
+		virtualMachineScaleSetTypeSb.WriteString("/")
+		virtualMachineScaleSetTypeSb.WriteString(constVirtualMachineScaleSetsResource)
+		for _, res := range resourceInfos {
```

Comment:
Instead of looping through the resources infos twice, both guarded by an if statement, what do you think about looping once and using a `switch` statement to decide which path to go down? I would further recommend refactoring out the logic inside each if-statement into a separate function so the logic is easier to read. What do you think?